### ISSUE TITLE
[BUG] set_golem_name_tests does not work as expected in set_golem_name

### ISSUE THREAD

**Issue by ColinFay** [ThinkR-open/golem#1161]
```r
golem::create_golem("pouet")
usethis::use_test()
golem::set_golem_name("pj") # calls set_golem_name_tests()
> readLines("tests/testthat.R") |> cat(sep = "\n")
# This file is part of the standard setup for testthat.
# It is recommended that you do not modify it.
#
# Where should you do additional test configuration?
# Learn more about the roles of various files in:
# * https://r-pkgs.org/testing-design.html#sec-tests-files-overview
# * https://testthat.r-lib.org/articles/special-files.html

library(testthat)
library(pouet) # Should be pj

test_check("pouet") # Should be pj
```

**Comment by ColinFay** [comment]
This is due to lazy evaluation. 

- We enter the function with `name = golem::pkg_name()` & `old_name = golem::pkg_name()` but they are not evaluated yet https://github.com/ThinkR-open/golem/blob/dev/R/golem-yaml-set.R#L31 
- Both these arguments, when they are evaluated, will run the fun & read the `DESCRIPTION` file to get the name of the package
- The name in `DESCRIPTION` is changed __before__ the `old_name` argument is evaluated https://github.com/ThinkR-open/golem/blob/dev/R/golem-yaml-set.R#L61
- Hence when we evaluate `old_name` it goes and read the DESCRIPTION after it has been changed to `name`
- Then the gsub() fails, as old_name == name

**Comment by ColinFay** [comment]
Closed via 0d0f679
